src/batch/src/task/task_execution.rs

```diff
@@ -285,12 +286,14 @@ impl<C: BatchTaskContext> BatchTaskExecution<C> {
                 {
                     // Prints the entire backtrace of error.
                     error!("Execution failed [{:?}]: {:?}", &task_id, &e);
+                    let err_str = e.to_string();
                     *failure.lock() = Some(e);
-                    if let Err(_e) = t_1
-                        .change_state_notify(TaskStatus::Failed, &mut state_tx)
+                    if let Err(_e) = self
+                        .change_state_notify(TaskStatus::Failed, &mut state_tx, Some(err_str))
```

Comment:
I think it's better to add a field `failure_reason` in `Failed` enum rather than adding an argument.

Reply:
Yes, but because TaskStatus is a struct from proto, it's hard to add this into that variant...